gridcomps/ExtData3G/AbstractDataSetFileSelector.F90

```diff
@@ -0,0 +1,148 @@
+#include "MAPL_Exceptions.h"
+#include "MAPL_ErrLog.h"
+module mapl3g_AbstractDataSetFileSelector
+   use ESMF
+   use MAPL_KeywordEnforcerMod
+   use MAPL_ExceptionHandling
+   use mapl3g_DataSetBracket
+   use mapl_StringTemplate
+   implicit none
+   private
+
+   public AbstractDataSetFileSelector
+   public file_not_found
+   public NUM_SEARCH_TRIES
+
+   integer, parameter :: MAX_TRIALS = 10
+   integer, parameter :: NUM_SEARCH_TRIES = 1
+   character(len=*), parameter :: file_not_found = "NONE"
+ 
+   type, abstract :: AbstractDataSetFileSelector
+      character(:), allocatable :: file_template
+      type(ESMF_TimeInterval)  :: frequency
```

Comment:
You may want a specific name than `frequency`.   Not immediately obvious how it is different than the clock dt.

Reply:
@tclune I changed to `file_frequency`